heudiconv/config.py

```diff
@@ -0,0 +1,188 @@
+"""
+A Python module for managing configurations across a heudiconv run.
+"""
+
+import atexit
+import os
+from pathlib import Path
+
+
+# avoid telemetry check if user desires
+_disable_et = bool(
+    os.getenv("NO_ET") is not None
+    or os.getenv("NIPYPE_NO_ET") is not None
+)
+os.environ["NIPYPE_NO_ET"] = "1"
+os.environ["NO_ET"] = "1"
+_latest_version = "Unknown"
+
+if not _disable_et:
+    # Ensure analytics hit only once
+    from contextlib import suppress
+    from requests import get, ConnectionError, ReadTimeout
+    with suppress((ConnectionError, ReadTimeout)):
+        res = get("https://rig.mit.edu/et/projects/nipy/heudiconv", timeout=0.5)
+    try:
+        _latest_version = res.json()['version']
+    except Exception:
+        pass
+
+
+class _Config:
+    """An abstract class forbidding instantiation."""
+
+    _paths = tuple()
+
+    def __init__(self):
+        """Avert instantiation."""
+        raise RuntimeError('Configuration type is not instantiable.')
```

Comment:
why not to start using `abc` and thus declare this as an `abc.abstractmethod`? according to https://docs.python.org/3/library/abc.html#abc.abstractmethod could be applied to `@classmethod` as well